tests/test_bookmarks.py

```diff
@@ -0,0 +1,125 @@
+from dateutil.parser import parse
+from base import AsanaBase
+import tap_tester.connections as connections
+import tap_tester.menagerie as menagerie
+import tap_tester.runner as runner
+
+class AsanaBookmarksTest(AsanaBase):
+
+    def name(self):
+        return "tap_tester_asana_bookmarks_test"
+
+    def test_run(self):
+        """
+        Testing that the bookmarking for the tap works as expected
+        - Verify for each incremental stream you can do a sync which records bookmarks
+        - Verify that a bookmark doesn't exist for full table streams.
+        - Verify the bookmark is the max value sent to the target for the a given replication key.
+        - Verify 2nd sync respects the bookmark
+        - All data of the 2nd sync is >= the bookmark from the first sync
+        - The number of records in the 2nd sync is less then the first
+        """
+        conn_id = connections.ensure_connection(self)
+        runner.run_check_mode(self, conn_id)
+
+        expected_streams = self.expected_streams()
+
+        found_catalogs = self.run_and_verify_check_mode(conn_id)
+        self.select_found_catalogs(conn_id, found_catalogs, only_streams=expected_streams)
+
+        # Run a sync job using orchestrator
+        first_sync_record_count = self.run_and_verify_sync(conn_id)
+        first_sync_records = runner.get_records_from_target_output()
+        first_sync_bookmarks = menagerie.get_state(conn_id)
+
+        ##########################################################################
+        ### Update State
+        ##########################################################################
+
+        # setting 'second_start_date' as bookmark for running 2nd sync
+        new_state = {'bookmarks': dict()}
+        replication_keys = self.expected_replication_keys()
+        for stream in expected_streams:
+            if self.is_incremental(stream):
+                new_state['bookmarks'][stream] = dict()
+                new_state['bookmarks'][stream][next(iter(replication_keys[stream]))] = self.second_start_date
```

Comment:
Could you either rename this variable here, or add an explanation explicitly calling out that this date is being re-used from the start date test because it is more recent than the default start date of used by the connection and is acting as a simulated bookmark. I want to make sure it is clear to future testers exactly what this test does, and I think mentioning start date could be confusing for anyone unfamiliar with the details of the test.

Reply:
Added a comment in the code.